### ISSUE TITLE
Rule Change: Update sort-keys-fix to sort numeric keys by value rather than alphabetical order

### ISSUE THREAD

**Issue by tlevesque-ueat** [eslint/eslint#15793]
### What rule do you want to change?

sort-keys-fix/sort-keys-fix

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
export const something = {
  [1]: 'foo',
  [2]: 'bar',
  [3]: 'baz',
  [10]: 'blah',
  [11]: 'hello',
};
```

### What does the rule currently do for this code?

This generate the following error:

```
5:4  error  Expected object keys to be in ascending order. '10' should be before '3'  sort-keys-fix/sort-keys-fix
```

### What will the rule do after it's changed?

It won't report an error. The keys are numbers here; they should be sorted by their value, not by alphabetical order.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @tlevesque-ueat, thanks for the issue!
 
The core `sort-keys` rule has option [`natural: true`](https://eslint.org/docs/rules/sort-keys#natural) for this behavior (here's a lint-free [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IHNvcnQta2V5czogW2Vycm9yLCBcImFzY1wiLCB7IG5hdHVyYWw6IHRydWUgfV0gKi9cblxuZXhwb3J0IGNvbnN0IHNvbWV0aGluZyA9IHtcbiAgWzFdOiAnZm9vJyxcbiAgWzJdOiAnYmFyJyxcbiAgWzNdOiAnYmF6JyxcbiAgWzEwXTogJ2JsYWgnLFxuICBbMTFdOiAnaGVsbG8nLFxufTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjoyLCJmb3ItZGlyZWN0aW9uIjoyLCJnZXR0ZXItcmV0dXJuIjoyLCJuby1hc3luYy1wcm9taXNlLWV4ZWN1dG9yIjoyLCJuby1jYXNlLWRlY2xhcmF0aW9ucyI6Miwibm8tY2xhc3MtYXNzaWduIjoyLCJuby1jb21wYXJlLW5lZy16ZXJvIjoyLCJuby1jb25kLWFzc2lnbiI6Miwibm8tY29uc3QtYXNzaWduIjoyLCJuby1jb25zdGFudC1jb25kaXRpb24iOjIsIm5vLWNvbnRyb2wtcmVnZXgiOjIsIm5vLWRlYnVnZ2VyIjoyLCJuby1kZWxldGUtdmFyIjoyLCJuby1kdXBlLWFyZ3MiOjIsIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6Miwibm8tZHVwZS1lbHNlLWlmIjoyLCJuby1kdXBlLWtleXMiOjIsIm5vLWR1cGxpY2F0ZS1jYXNlIjoyLCJuby1lbXB0eSI6Miwibm8tZW1wdHktY2hhcmFjdGVyLWNsYXNzIjoyLCJuby1lbXB0eS1wYXR0ZXJuIjoyLCJuby1leC1hc3NpZ24iOjIsIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6Miwibm8tZXh0cmEtc2VtaSI6Miwibm8tZmFsbHRocm91Z2giOjIsIm5vLWZ1bmMtYXNzaWduIjoyLCJuby1nbG9iYWwtYXNzaWduIjoyLCJuby1pbXBvcnQtYXNzaWduIjoyLCJuby1pbm5lci1kZWNsYXJhdGlvbnMiOjIsIm5vLWludmFsaWQtcmVnZXhwIjoyLCJuby1pcnJlZ3VsYXItd2hpdGVzcGFjZSI6Miwibm8tbG9zcy1vZi1wcmVjaXNpb24iOjIsIm5vLW1pc2xlYWRpbmctY2hhcmFjdGVyLWNsYXNzIjoyLCJuby1taXhlZC1zcGFjZXMtYW5kLXRhYnMiOjIsIm5vLW5ldy1zeW1ib2wiOjIsIm5vLW5vbm9jdGFsLWRlY2ltYWwtZXNjYXBlIjoyLCJuby1vYmotY2FsbHMiOjIsIm5vLW9jdGFsIjoyLCJuby1wcm90b3R5cGUtYnVpbHRpbnMiOjIsIm5vLXJlZGVjbGFyZSI6Miwibm8tcmVnZXgtc3BhY2VzIjoyLCJuby1zZWxmLWFzc2lnbiI6Miwibm8tc2V0dGVyLXJldHVybiI6Miwibm8tc2hhZG93LXJlc3RyaWN0ZWQtbmFtZXMiOjIsIm5vLXNwYXJzZS1hcnJheXMiOjIsIm5vLXRoaXMtYmVmb3JlLXN1cGVyIjoyLCJuby11bmRlZiI6Miwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOjIsIm5vLXVucmVhY2hhYmxlIjoyLCJuby11bnNhZmUtZmluYWxseSI6Miwibm8tdW5zYWZlLW5lZ2F0aW9uIjoyLCJuby11bnNhZmUtb3B0aW9uYWwtY2hhaW5pbmciOjIsIm5vLXVudXNlZC1sYWJlbHMiOjIsIm5vLXVudXNlZC12YXJzIjoyLCJuby11c2VsZXNzLWJhY2tyZWZlcmVuY2UiOjIsIm5vLXVzZWxlc3MtY2F0Y2giOjIsIm5vLXVzZWxlc3MtZXNjYXBlIjoyLCJuby13aXRoIjoyLCJyZXF1aXJlLXlpZWxkIjoyLCJ1c2UtaXNuYW4iOjIsInZhbGlkLXR5cGVvZiI6Mn0sImVudiI6e319fQ==) with your example), but you're actually using a plugin rule `sort-keys-fix/sort-keys-fix`, so this is rather a question for plugin's repository:

https://github.com/leo-buneev/eslint-plugin-sort-keys-fix

I'm closing this issue since it isn't related to ESLint core rules.

(per the documentation for `sort-keys-fix/sort-keys-fix`, it seems that it supports the same `natural: true` option, so configuring the rule with `["error", "asc", {natural: true}]` would probably make it behave the way you want).

**Comment by tlevesque-ueat** [comment]
> you're actually using a plugin rule

Ah, sorry, I didn't realize. I'm kinda new to the JS ecosystem...

> (per the documentation for `sort-keys-fix/sort-keys-fix`, it seems that it supports the same `natural: true` option, so configuring the rule with `["error", "asc", {natural: true}]` would probably make it behave the way you want).

Thanks!
